hadoop-yarn-project/hadoop-yarn/hadoop-yarn-server/hadoop-yarn-server-resourcemanager/src/test/java/org/apache/hadoop/yarn/server/resourcemanager/CommonUtil.java

```diff
@@ -0,0 +1,410 @@
+                .getApplicationAttempt(attemptId) && tick < 50) {
+            Thread.sleep(100);
+            if (tick % 10 == 0) {
+                LOG.info("waiting for SchedulerApplicationAttempt="
+                        + attemptId + " added.");
+            }
+            tick++;
+        }
+        Assert.assertNotNull("Timed out waiting for SchedulerApplicationAttempt=" +
+                attemptId + " to be added.", ((AbstractYarnScheduler)
+                rm.getResourceScheduler()).getApplicationAttempt(attemptId));
+    }
+
+    public static RMAppAttempt waitForAttemptScheduled(RMApp app, MockRM rm)
+            throws Exception {
+        waitForState(rm, app.getApplicationId(), RMAppState.ACCEPTED);
+        RMAppAttempt attempt = app.getCurrentAppAttempt();
+        waitForSchedulerAppAttemptAdded(attempt.getAppAttemptId(), rm);
+        waitForState(rm, attempt.getAppAttemptId(), RMAppAttemptState.SCHEDULED);
+        return attempt;
+    }
+
+    /**
+     * Wait until an app removed from scheduler.
+     * The timeout is 40 seconds.
+     * @param appId the id of an app
+     * @throws InterruptedException
+     *         if interrupted while waiting for app removed
+     */
+    public static void waitForAppRemovedFromScheduler(MockRM rm, ApplicationId appId)
+            throws InterruptedException {
+        int timeWaiting = 0;
+        rm.drainEventsImplicitly();
+        Map<ApplicationId, SchedulerApplication> apps  =
+                ((AbstractYarnScheduler) rm.getResourceScheduler())
+                        .getSchedulerApplications();
+        while (apps.containsKey(appId)) {
+            if (timeWaiting >= TIMEOUT_MS_FOR_APP_REMOVED) {
+                break;
+            }
+            LOG.info("wait for app removed, " + appId);
+            Thread.sleep(WAIT_MS_PER_LOOP);
+            timeWaiting += WAIT_MS_PER_LOOP;
+        }
+        Assert.assertTrue("app is not removed from scheduler (timeout).",
+                !apps.containsKey(appId));
+        LOG.info("app is removed from scheduler, " + appId);
+    }
+
+    /**
+     * Wait until a container has reached a completion state.
+     * @param rm             A mock resourcemanager
+     * @param nm             A mock nodemanager
+     * @param amContainerId  A containerId for AM
+     * @param container      Resourcemanager's view of an application container
+     */
+    public static void waitforContainerCompletion(MockRM rm, MockNM nm,
+                                            ContainerId amContainerId, RMContainer container) throws Exception {
+        ContainerId containerId = container.getContainerId();
+        if (null != rm.scheduler.getRMContainer(containerId)) {
+            if (containerId.equals(amContainerId)) {
+                CommonUtil.waitForState(rm, nm, containerId, RMContainerState.COMPLETED);
+            } else {
+                CommonUtil.waitForState(rm, nm, containerId, RMContainerState.KILLED);
+            }
```

Comment:
I think this can be replace with a ternary operator, so we can reduce the code here.
```
finalState = containerId.equals(amContainerId) ? RMContainerState.COMPLETED : RMContainerState.KILLED
CommonUtil.waitForState(rm, nm, containerId, finalState)
```